R-proj/src/rounding.cpp

```diff
@@ -64,14 +72,40 @@ Rcpp::List rounding (Rcpp::Reference P, Rcpp::Nullable<double> seed = R_NilValue
         walkL = 2;
     }
 
+    MT N = MT::Identity(n,n);
+    VT N_shift = VT::Zero(n);
+    NT svd_prod = 1.0;
     switch (type) {
         case 1: {
             // Hpolytope
-            HP.init(n, Rcpp::as<MT>(P.field("A")), Rcpp::as<VT>(P.field("b")));
+            if (Rcpp::as<MT>(P.field("Aeq")).rows() == 0) {
+                // full dimensional
+                HP.init(n, Rcpp::as<MT>(P.field("A")), Rcpp::as<VT>(P.field("b")));
+            } else {
+                // low dimensional
+                std::pair<Hpolytope, std::pair<MT, VT> > temp_res = get_full_dimensional_polytope<Hpolytope>(Rcpp::as<MT>(P.field("A")), Rcpp::as<VT>(P.field("b")),
+                            Rcpp::as<MT>(P.field("Aeq")), Rcpp::as<VT>(P.field("beq")));
+
+                HP = temp_res.first;
+                N = temp_res.second.first;
+                N_shift = temp_res.second.second;
+
+                Eigen::JacobiSVD<MT> svd(N, Eigen::ComputeThinU | Eigen::ComputeThinV);
```

Comment:
Do you need here the computations of Eigen::ComputeThinU | Eigen::ComputeThinV ?

Reply:
The computation of U is not needed. I removed it.